enterprise_access/apps/api_client/enterprise_catalog_client.py

```diff
@@ -32,3 +32,37 @@ def contains_content_items(self, catalog_uuid, content_ids):
         response.raise_for_status()
         response_json = response.json()
         return response_json.get('contains_content_items', False)
+
+    def catalog_content_metadata(self, catalog_uuid, content_keys, traverse_pagination=True, **kwargs):
+        """
+        Returns a list of requested content metadata records for the given catalog_uuid.
+        See the enterprise-catalog ``EnterpriseCatalogGetContentMetadata`` view.
+
+        Arguments:
+            catalog_uuid (UUID): UUID of the enterprise catalog to check.
+            content_keys (list of str): List of content keys in the catalog for which metadata should be fetched.
+            traverse_pagination (bool, default True): If true, forces the requested endpoint
+                to tranverse pagination for us.
+                This means a single response payload will contain all results
+                and there's no need for us to fetch multiple pages.
+
+        Returns:
+            A paginated results dict, where the "results" key contains
+            a list of dicts. These are content metadata dicts for the requested keys
+            (as long as they are associated with the given catalog_uuid).
+            When the "next" key of results is not null, there are further
+            pages of results that can be fetched - it's up to the caller to fetch these.
+        """
+        if not content_keys and traverse_pagination:
+            raise Exception('Cannot request all metadata for a catalog when traverse_pagination is true.')
+
+        query_params = {
+            'content_keys': content_keys,
```

Comment:
[observation] Will be interesting to see how many assigments learners end up having, and how quickly (if at all) we might run up against the `MAX_GET_CONTENT_KEYS=100` set for this API.

Reply:
ah good point, I'll note that limit in the docstring